ci/scripts/msys2_setup.sh

```diff
@@ -72,15 +71,6 @@ case "${target}" in
     ;;
 esac
 
-case "${target}" in
-  cgo)
-    packages+=(${MINGW_PACKAGE_PREFIX}-arrow)
-    packages+=(${MINGW_PACKAGE_PREFIX}-gcc)
-    packages+=(${MINGW_PACKAGE_PREFIX}-toolchain)
-    packages+=(base-devel)
-    ;;
-esac
-
```

Comment:
This is unrelated to this fix. Sorry.

We don't need cgo related code because we move Go related codes to apache/arrow-go.